x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PruneColumns.java

```diff
@@ -148,24 +96,143 @@ public LogicalPlan apply(LogicalPlan plan) {
         return pl;
     }
 
+    private static LogicalPlan pruneColumns(Aggregate aggregate, AttributeSet.Builder used, boolean inlineJoin) {
+        LogicalPlan p = aggregate;
+
+        var remaining = pruneUnusedAndAddReferences(aggregate.aggregates(), used);
+
+        if (remaining != null) {
```

Comment:
What if you checked here `if (remaining == null) return p;`? to have a smaller `if` branch?